addons/event/static/src/client_action/event_barcode.js

```diff
@@ -0,0 +1,111 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { BarcodeScanner } from "@barcodes/components/barcode_scanner";
+import { Component, onWillStart } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { useBus, useService } from "@web/core/utils/hooks";
+import { EventRegistrationSummaryDialog } from "./event_registration_summary_dialog";
+
+export class EventScanView extends Component {
+    setup() {
+        this.actionService = useService("action");
+        this.dialog = useService("dialog");
+        this.notification = useService("notification");
+        this.orm = useService("orm");
+        this.rpc = useService("rpc");
+
+        const { default_event_id, active_model, active_id } = this.props.action.context;
+        this.eventId = default_event_id || (active_model === "event.event" && active_id);
+        this.isMultiEvent = !this.eventId;
+
+        const barcode = useService("barcode");
+        useBus(barcode.bus, "barcode_scanned", (ev) => this.onBarcodeScanned(ev.detail.barcode));
+
+        onWillStart(this.onWillStart);
+    }
+
+    /**
+     * @override
+     * Fetch barcode init information. Notably eventId triggers mono- or multi-
+     * event mode (Registration Desk in multi event allow to manage attendees
+     * from several events and tickets without reloading / changing event in UX.
+     */
+    async onWillStart() {
+        this.data = await this.rpc("/event/init_barcode_interface", {
```

Comment:
Could be renamed to eventData, to ease understanding its usage in the file :)

Reply:
@tde-banana-odoo I think changing the name won't work, since this method overrides the parent